pkg/dag/graph.go

```diff
@@ -262,43 +262,58 @@ func (g *Graph) addAppropriatePackage(resolver *apko.PkgResolver, c Package, dep
 // It discovers what the current dependency is that is causing the potential loop,
 // removes the last edge in that cycle, and regenerates that dependency without the previous target.
 func (g *Graph) resolveCycle(c *cycle, dep string, resolver *apko.PkgResolver, localRepoSource string) error {
```

Comment:
Would it be feasible to add a test for the expectation for how cycle resolution works?

My impression is that the ability to resolve cycles like this is important, and I'm nervous about myself not being able to absorb all the logical context here fast enough before I submit a change that breaks something. Having a test would help "future me", both in that it'd give examples of what was expected to happen, and in that it'd ensure I didn't break an important aspect here due to my own lack of competence in this area!

Reply:
It is a good idea to add them. I usually am strongly in favour. At this point, to be honest, I struggled with how to create a good test for this (which just testifies to the complexity). I spent quite some time collecting the fail case and creating the attached issue just to begin to understand what was happening.

We can try, though

Reply:
Okay! I can approve for now (I paused just because I know auto-merge can hide comments sometimes).

But just something to think about, and noting my current level of nervousness (maybe a 5 out of 10 😅 )

Reply:
No, I would like to actually tackle it. it is important. It cannot really be unit-tested, though, as it involves so much more than just `resolveCycle()`. It cleans some things up and then redoes calls.

We might be able to find a way to test parts of it, and then maybe integration-test the whole thing.

Reply:
That makes sense, I figured this might be a higher-level, more-context kind of test

Reply:
I think I might have figured out a decent test for this.

Reply:
Have a look at #236 @luhring